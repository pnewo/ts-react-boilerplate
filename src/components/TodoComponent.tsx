import * as React from 'react';
import Todo from '../common/Todo';

export interface ITodoComponent {
    readonly todo: Todo;
    setDone(i: number);
}

const TodoComponent: React.StatelessComponent<ITodoComponent> = ({ todo, setDone }) => (
    <div className="todo">
        <input
            className="todo__checkbox"
            type="checkbox"
            checked={todo.done}
            onChange={() => !todo.done && setDone(todo.id)}
        />
        <span className="todo__number">{todo.id}:</span>
        <span className="todo__title">{todo.title}</span>
    </div>
);

export default TodoComponent;
