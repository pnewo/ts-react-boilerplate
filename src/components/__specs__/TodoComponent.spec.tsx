import * as React from 'react';
import { shallow } from 'enzyme';
import TodoComponent from '../TodoComponent';
import Todo from '../../common/Todo';

describe('TodoComponent', () => {
    const testTodo1 = new Todo(1, 'Title');
    const testTodo2 = new Todo(0, 'Testing', true);
    const setDone = jest.fn();
    const wrapper1 = shallow(<TodoComponent todo={testTodo1} setDone={setDone} />);
    const wrapper2 = shallow(<TodoComponent todo={testTodo2} setDone={setDone} />);

    it('should render correctly', () => {
        expect(wrapper1).toMatchSnapshot();
        expect(wrapper2).toMatchSnapshot();
    });

    it('should call setDone correctly', () => {
        wrapper1.find('.todo__checkbox').simulate('change');
        expect(setDone).toBeCalledWith(testTodo1.id);
        expect(setDone).toHaveBeenCalledTimes(1);
        wrapper2.find('.todo__checkbox').simulate('change');
        expect(setDone).toHaveBeenCalledTimes(1);
    });
});
