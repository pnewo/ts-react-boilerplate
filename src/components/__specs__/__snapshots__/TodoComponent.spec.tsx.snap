// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TodoComponent should render correctly 1`] = `
<div
  className="todo"
>
  <input
    checked={false}
    className="todo__checkbox"
    onChange={[Function]}
    type="checkbox"
  />
  <span
    className="todo__number"
  >
    1
    :
  </span>
  <span
    className="todo__title"
  >
    Title
  </span>
</div>
`;

exports[`TodoComponent should render correctly 2`] = `
<div
  className="todo"
>
  <input
    checked={true}
    className="todo__checkbox"
    onChange={[Function]}
    type="checkbox"
  />
  <span
    className="todo__number"
  >
    0
    :
  </span>
  <span
    className="todo__title"
  >
    Testing
  </span>
</div>
`;
